pkg/agent/servicecidr/discoverer.go

```diff
@@ -64,7 +75,38 @@ func NewServiceCIDRDiscoverer(serviceInformer coreinformers.ServiceInformer) Int
 	return d
 }
 
-func (d *discoverer) GetServiceCIDR(isIPv6 bool) (*net.IPNet, error) {
+func (d *Discoverer) Run(stopCh <-chan struct{}) {
+	defer d.queue.ShutDown()
+
+	klog.Info("Starting ServiceCIDRDiscoverer")
+	defer klog.Info("Stopping ServiceCIDRDiscoverer")
+	if !cache.WaitForCacheSync(stopCh, d.serviceInformer.HasSynced) {
+		return
+	}
+	svcs, _ := d.serviceLister.List(labels.Everything())
+	d.updateServiceCIDR(svcs...)
```

Comment:
Here all existing Services are listed, and a calculated Service CIDR will be got, how could we clean the stale Service route? Previously, we use the first ClusterIP to collect all routes whose destination includes the ClusterIP.

Reply:
Previous cleanup logic is not reliable, there is no guarantee the first clusterIP has been covered the previous routes. Please check the latest code about how stale routes are collected.